words/assembler.asm

```diff
@@ -24,962 +25,76 @@
-z_asm_sta_zi:
-
-xt_asm_sta_ziy: ; sta.ziy \ STA (nn),Y
-                lda #$91
-                bra asm_common
-z_asm_sta_ziy:
-
-xt_asm_sta_zx:  ; sta.zx \ STA nn,X
-                lda #$95
-                bra asm_common
-z_asm_sta_zx:
-
-xt_asm_sta_zxi: ; sta.zxi \ STA (nn,X)
-                lda #$81
-                bra asm_common
-z_asm_sta_zxi:
-
-xt_asm_stx:     ; stx \ STX nnnn
-                lda #$8E
-                bra asm_common
-z_asm_stx:
-
-xt_asm_stx_z:   ; stx.z \ STX nn
-                lda #$86
-                bra asm_common
-z_asm_stx_z:
-
-xt_asm_stx_zy:  ; stx.zy \ STX nn,Y
-                lda #$96
-                bra asm_common
-z_asm_stx_zy:
-
-xt_asm_sty:     ; sty \ STY nnnn
-                lda #$8C
-                bra asm_common
-z_asm_sty:
-
-xt_asm_sty_z:   ; sty.z \ STY nn
-                lda #$84
-                bra asm_common
-z_asm_sty_z:
-
-xt_asm_sty_zx:  ; sty.zx \ STY nn,X
-                lda #$94
-                bra asm_common
-z_asm_sty_zx:
-
-xt_asm_stz:     ; stz \ STZ nnnn
-                lda #$9C
-                bra asm_common
-z_asm_stz:
-
-xt_asm_stz_x:   ; stz.x \ STZ nnnn,X
-                lda #$9E
-                bra asm_common
-z_asm_stz_x:
-
-xt_asm_stz_z:   ; stz.z \ STZ nn
-                lda #$64
-                bra asm_common
-z_asm_stz_z:
-
-xt_asm_stz_zx:  ; stz.zx \ STZ nn,X
-                lda #$74
-                bra asm_common
-z_asm_stz_zx:
-
-xt_asm_tax:     ; tax \ TAX
-                lda #$AA
-                bra asm_common
-z_asm_tax:
-
-xt_asm_tay:     ; tay \ TAY
-                lda #$A8
-                bra asm_common
-z_asm_tay:
-
-xt_asm_trb:     ; trb \ TRB nnnn
-                lda #$1C
-                bra asm_common
-z_asm_trb:
-
-xt_asm_trb_z:   ; trb.z \ TRB nn
-                lda #$14
-                bra asm_common
-z_asm_trb_z:
-
-xt_asm_tsb:     ; tsb \ TSB nnnn
-                lda #$0C
-                bra asm_common
-z_asm_tsb:
-
-xt_asm_tsb_z:   ; tsb.z \ TSB nn
-                lda #$04
-                bra asm_common
-z_asm_tsb_z:
-
-xt_asm_tsx:     ; tsx \ TSX
-                lda #$BA
-                bra asm_common
-z_asm_tsx:
-
-xt_asm_txa:     ; txa \ TXA
-                lda #$8A
-                bra asm_common
-z_asm_txa:
-
-xt_asm_txs:     ; txs \ TXS
-                lda #$9A
-                bra asm_common
-z_asm_txs:
-
-xt_asm_tya:     ; tya \ TYA
-                lda #$98
-                bra asm_common
-z_asm_tya:
-
 
 ; ==========================================================
 ; ASSEMBLER HELPER FUNCTIONS
 
-asm_common:
+; This table repeats `jsr asm_op_common` 256 times, each 3 bytes long.
+; Because 3 and 256 are relatively prime, for any opcode op
+; there will be exactly one such entrypoint which generates a return
+; address (at entrypoint+2) whose LSB matches op
+
+asm_op_table:
+.rept 256
+        jsr asm_op_common
+.endrept
+
+; This macro calculates the relevant entrypoint from the table.
```

Comment:
this is the trick to pick the right XT for a given opcode, regardless of where the table of 256 jsr's lands.